x/auctions/keeper/msg_server_test.go

```diff
@@ -0,0 +1,387 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper_test
+
+import (
+	"testing"
+
+	testkeyring "github.com/evmos/evmos/v19/testutil/integration/evmos/keyring"
+	testnetwork "github.com/evmos/evmos/v19/testutil/integration/evmos/network"
+	testutiltx "github.com/evmos/evmos/v19/testutil/tx"
+	"github.com/stretchr/testify/assert"
+
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/evmos/evmos/v19/utils"
+	"github.com/evmos/evmos/v19/x/auctions/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestBid(t *testing.T) {
+	var network *testnetwork.UnitTestNetwork
+
+	// validSenderKey is populated during each
+	// run and used to create the MsgBid.
+	var validSenderKey testkeyring.Key
+	bidAmount := sdk.NewInt(5)
+	emptyAddress, _ := testutiltx.NewAccAddressAndKey()
+
+	testCases := []struct {
+		name        string
+		malleate    func()               // used to modify the initial conditions of the test
+		input       func() *types.MsgBid // return the message used for the bid
+		postCheck   func()
+		expErr      bool
+		errContains string
+	}{
+		{
+			name: "pass with no previous bid",
+			malleate: func() {
+			},
+			input: func() *types.MsgBid {
+				return &types.MsgBid{
+					Sender: validSenderKey.AccAddr.String(),
+					Amount: sdk.NewCoin(utils.BaseDenom, bidAmount),
+				}
+			},
+			postCheck: func() {},
+			expErr:    false,
+		},
+		{
+			name: "pass with previous bid present",
+			malleate: func() {
+				// Send coins from the valid sender to an empty account. In this
+				// way we can easily verify the expected final balance.
+				emptyAccountCoin := sdk.NewCoin(utils.BaseDenom, bidAmount.Sub(sdk.NewInt(1)))
+				err := network.App.BankKeeper.SendCoins(network.GetContext(), validSenderKey.AccAddr, emptyAddress, sdk.NewCoins(emptyAccountCoin))
+				assert.NoError(t, err, "failed to send coins from valid sender to empty account")
+				bigMsg := &types.MsgBid{
+					Sender: emptyAddress.String(),
+					Amount: emptyAccountCoin,
+				}
+				_, err = network.App.AuctionsKeeper.Bid(network.GetContext(), bigMsg)
+				assert.NoError(t, err, "failed to create setup bid")
+			},
+			input: func() *types.MsgBid {
+				return &types.MsgBid{
+					Sender: validSenderKey.AccAddr.String(),
+					Amount: sdk.NewCoin(utils.BaseDenom, bidAmount),
+				}
+			},
+			postCheck: func() {
+				resp := network.App.BankKeeper.GetBalance(network.GetContext(), emptyAddress, utils.BaseDenom)
+				assert.Equal(t, resp.Amount, bidAmount.Sub(sdk.NewInt(1)))
```

Comment:
would be nicer to create a `var` here to not run into any problems in the future and to make it more readable


```suggestion
				assert.Equal(t, resp.Amount, bidAmount.Sub(previousBidAmount))
```